src/Query/DateParser.ts

```diff
@@ -75,12 +93,74 @@ export class DateParser {
+            DateParser.setDateRangeToStartOfDay(dateRange);
 
             return dateRange;
         }
 
         return undefined;
     }
+
+    private static specificYearRegex = /[0-9]{4}/;
+    private static specificQuarterRegex = /[0-9]{4}-Q[1-4]/;
+    private static specificMonthRegex = /[0-9]{4}-[0-9]{2}/;
+    private static specificWeekRegex = /[0-9]{4}-W[0-9]{2}/;
+    private static specificYearFormat = 'YYYY';
+    private static specificQuarterFormat = 'YYYY-Q';
+    private static specificMonthFormat = 'YYYY-MM';
+    private static specificWeekFormat = 'YYYY-WW';
+
+    private static parseSpecificDateRange(input: string): [moment.Moment, moment.Moment] | undefined {
+        const yearMatch = input.match(DateParser.specificYearRegex);
+        if (yearMatch && yearMatch.length === 1 && yearMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(yearMatch[0], DateParser.specificYearFormat);
+        }
+
+        const quarterMatch = input.match(DateParser.specificQuarterRegex);
+        if (quarterMatch && quarterMatch.length === 1 && quarterMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(quarterMatch[0], DateParser.specificQuarterFormat);
+        }
+
+        const monthMatch = input.match(DateParser.specificMonthRegex);
+        if (monthMatch && monthMatch.length === 1 && monthMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(monthMatch[0], DateParser.specificMonthFormat);
+        }
+
+        const weekMatch = input.match(DateParser.specificWeekRegex);
+        if (weekMatch && weekMatch.length === 1 && weekMatch[0] === input) {
+            return DateParser.buildSpecificDateRange(weekMatch[0], DateParser.specificWeekFormat);
+        }
+
+        return undefined;
+    }
+
+    private static buildSpecificDateRange(range: string, format: string): [moment.Moment, moment.Moment] {
+        let unit: moment.unitOfTime.Base | moment.unitOfTime._quarter | moment.unitOfTime._isoWeek = 'second';
+        switch (format) {
+            case DateParser.specificYearFormat:
+                unit = 'year';
+                break;
+            case DateParser.specificQuarterFormat:
+                unit = 'quarter';
+                break;
+            case DateParser.specificMonthFormat:
+                unit = 'month';
+                break;
+            case DateParser.specificWeekFormat:
+                unit = 'isoWeek';
+                break;
+        }
+
+        const dateRange: [moment.Moment, moment.Moment] = [
+            moment(range, format).startOf(unit),
+            moment(range, format).endOf(unit),
+        ];
+        DateParser.setDateRangeToStartOfDay(dateRange);
+        return dateRange;
+    }
+
+    private static setDateRangeToStartOfDay(dateRange: [moment.Moment, moment.Moment]) {
+        // Dates shall be at midnight eg 00:00
+        dateRange.forEach((d) => d.startOf('day'));
+    }
```

Comment:
Nice...

A small point: if this returned the adjusted range, the calls could all go from this:

```ts
            DateParser.setDateRangeToStartOfDay(dateRange);
            return dateRange;
```

to this:

```ts
            return DateParser.setDateRangeToStartOfDay(dateRange);
```

Reply:
ah, great!